grammar/fr/src/rules.rs

```diff
@@ -230,6 +242,15 @@ pub fn rules_duration(b: &mut RuleSetBuilder<Dimension>) -> RustlingResult<()> {
                  Ok(DurationValue::new(hour_period + minute_period))
              }
     );
+    b.rule_3("<integer> <unit-of-duration> et quart",
+        integer_check_by_range!(0),
+        unit_of_duration_check!(),
+        b.reg(r#"et quart"#)?,
+        |integer, uod, _| {
+           let quarter_period: Period = uod.value().grain.quarter_period().map(|a| a.into()).unwrap_or_else(|| Period::default());
+           Ok(DurationValue::new(quarter_period + PeriodComp::new(uod.value().grain, integer.value().value)))
```

Comment:
let quarter_period: Period = uod.value().grain.quarter_period()?.map_err(|_| RuleError::Invalid.into())?.into(). If no quarter period is possible, the rule won't be executed because it will raise an invalid error. 

WDYT ?

Reply:
Some cases were not taken into account, like "dans une heure et quart" (the resolution was 13:15:00)
So I duplicated the rule for "half". 
It's not OK ?